src/integ_test_resources/common/scripts/test/device_config_builder_test.py

```diff
@@ -68,6 +68,26 @@ def test_build_package_data_with_nesting(self):
         )
 
 
+    def test_get_credential_data(self):
```

Comment:
I might put some time into ensuring we don't litter a test run's environment by either [patching](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict) `os.environ` or ensuring these keys  are returned to their original state. It doesn't matter with the current tests, but could matter in the future.

Reply:
That's a really nice callout, thanks! Will add in an upcoming PR